### PR TITLE
Add BYO Certs Support and Support Rancher 2.6+ Bootstrapping for Control-Plane Module

### PR DESCRIPTION
**Pull request by git-ival**

### CODE REVIEW

File: control-plane/db.tf [hunk 1]
```diff
@@ -19,32 +19,34 @@ module "db" {
...
+  vpc_security_group_ids  = [data.aws_security_group.default.id, aws_security_group.database.id]
+  multi_az                = local.db_multi_az
+  deletion_protection     = false
 
   name     = var.db_name
   username = var.db_username
   password = var.db_password
   port     = var.db_port
-
+
```
Comment (rmweir) [inline control-plane/db.tf:42]: Unnecessary whitespace.

File: control-plane/modules/aws-k3s/files/register-to-rancher.sh [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 %{ if is_k3s_server }
   %{ if !install_rancher }
+echo "\'${registration_command}\'"
```
Comment (rmweir) [inline control-plane/modules/aws-k3s/files/register-to-rancher.sh:5]: Maybe prepend this with "Registration command: ". That way if it's empty or something we know what it was supposed to be.

File: control-plane/outputs.tf [hunk 3]
```diff
@@ -40,8 +40,8 @@ output "rancher_url" {
 }
 
 output "rancher_token" {
-  value     = module.k3s.rancher_token
-  sensitive = true
+  value     = var.sensitive_token ? null : nonsensitive(module.k3s.rancher_token)
+  sensitive = false
```
Comment (rmweir) [inline control-plane/outputs.tf:44]: Why has this changed?

File: control-plane/outputs.tf [hunk 4]
```diff
@@ -53,3 +53,52 @@ output "k3s_cluster_secret" {
...
+  value     = var.rancher_chart_tag
+}
+
+output "rancher_version" {
+  value     = var.rancher_version
+}
+
+output "use_new_bootstrap" {
+  value = module.k3s.use_new_bootstrap
+}
```
Comment (rmweir) [inline control-plane/outputs.tf:104]: Is the formatting correct on all of these? Looks awkward that the single line ones have additional whitespace.

File: control-plane/variables.tf [hunk 5]
```diff
@@ -136,4 +207,10 @@ variable "r53_domain" {
...
   description = "DNS domain for Route53 zone (defaults to domain if unset)"
-}
\ No newline at end of file
+}
+
+variable "sensitive_token" {
+  type = bool
+  default = true
+  description = "Boolean that determines if the module should treat the generated Rancher Admin API Token as sensitive in the output."
+}
```
Comment (rmweir) [inline control-plane/variables.tf:216]: What is the benefit of making this configurable?

**Review by rmweir — CHANGES_REQUESTED** [review]

File: control-plane/outputs.tf (hunk 3 above)
Comment (git-ival) [reply control-plane/outputs.tf:44]: Being able to mark this token as sensitive/nonsensitive allows for it to be displayed in the terraform output which is useful when using the `clusters` module to provision downstreams. It speeds up the manual process of getting an api token and allows for some shell-based automation.

File: control-plane/outputs.tf (hunk 4 above)
Comment (git-ival) [reply control-plane/outputs.tf:104]: Fair, I'll fix it. It was likely done by the terraform extension in vscode.

File: control-plane/variables.tf (hunk 5 above)
Comment (git-ival) [reply control-plane/variables.tf:216]: Same reasoning as the change to `module.k3s.rancher_token` in `control-plane/outputs.tf`. See my comment on that change for details.

**Review by rmweir — APPROVED** [review]
LGTM
